### DISCUSSION TITLE
For SWT checkboxes we see the color turning to blue on mousehover. Is there a way to replicate similar behavior with FocusIn listeners?

### DISCUSSION THREAD

**Discussion by purnac123** [eclipse-platform/eclipse.platform.swt#241] [General]
For SWT checkboxes we see the color turning to blue on mouse hover. 
Is there a way to replicate similar behavior with Focus In listeners? 
We have checkboxes in our application with no text assigned to them.
As a result when they have focus no dotted lines gets drawn around them as this is a known SWT issue.
As a workaround we want to highlight the checkbox blue on focus similar to what happens when we move our mouse over the checkbox.
Is there a way we can achieve this?
Sending **control.notifyListeners(SWT.MouseMove, new Event());**  or **control.notifyListeners(SWT.MouseMove, new Event());** on focus in does not seem to work for us.

**Comment by mickaelistria** [comment]
> For SWT checkboxes we see the color turning to blue on mouse hover.

The SWT checkboxes are delegated to native widgets on all OSs, so this is most likely the default behavior of your OS, not something SWT implements specifically.

> As a result when they have focus no dotted lines gets drawn around them as this is a known SWT issue.

I imagine it's also how native checkboxes are expected to behave. If not, is this reported as a bug in some tracker?

**Reply by purnac123** [reply]
Hi @mickaelistria ,

Thank you so much for your reply. 
For the issue of no dotted lines gets drawn around checkboxes with blank text I was able to find the below fix:
https://bugs.eclipse.org/bugs/show_bug.cgi?id=314815

Can you give me an idea as to how I can update my SWT project to the latest jar?
I see 4.25.0-SNAPSHOT is the latest snapshot version in https://github.com/eclipse-platform/eclipse.platform.swt.
However when I clone the repository and try running JavaViewer class from org.eclipse.swt.examples I see the below error message in POM:

Non-resolvable parent POM for eclipse.platform.swt:eclipse.platform.swt:4.25.0-SNAPSHOT: Could not find artifact org.eclipse:eclipse-platform-parent:pom:4.25.0-SNAPSHOT and 'parent.relativePath' points at wrong local POM.

The Button.class showing up as a plugin dependency is also displaying an older version of SWT jar, SWT.org.eclipse.swt.win32.win32.x86_64_3.114.100.v20200604-0951.jar and is not showing the changes that were part of the fix https://github.com/eclipse-platform/eclipse.platform.swt. or even the latest Button .java file for that matter.

If possible can you please let me know how to access the latest Button.java file?
